### PR TITLE
[ROCKETMQ-231] fix pull consumer pull result size

### PR DESCRIPTION
**Pull request by lindzh**
When using PullConsumer pull message by default result size is 32,and messages is more than 32 in a queue,but broker always returns 31.

### CODE REVIEW

File: test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import org.apache.rocketmq.test.base.IntegrationTestBase;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.channels.Pipe;
+import java.util.*;
+
+public class PullSizeTest extends BaseConf {
+
+    private static Logger logger = Logger.getLogger(PullSizeTest.class);
```
Comment (vongosling) [inline test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java]: Could we remove logger in unit test ?

File: test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                if (sendResult.getSendStatus().equals(SendStatus.SEND_OK)) {
+                    successCount++;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+                Thread.sleep(500);
+            }
+        }
+        producer.shutdown();
+        if (successCount > 800) {
```
Comment (vongosling) [inline test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java]: why use this magic number 800 in your if statement

File: test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                    default:
+                        break;
+                }
+            } catch (Exception e) {
+                logger.error("[SEND] send failed", e);
+                e.printStackTrace();
+            }
+        }
+        consumer.shutdown();
+        Assert.assertTrue(result);
```
Comment (vongosling) [inline test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java]: Could we assert the pull size is what you want, while not the simple true or false ?

File: test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java (hunk 1 above)
Comment (zhouxinyu) [inline test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java]: IMO, we don't need a separate it test file to cover this point, use unit test.

Comment (Jaskey) [inline test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java]: @zhouxinyu aggree

File: test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java (hunk 2 above)
Comment (lindzh) [inline test/src/test/java/org/apache/rocketmq/test/client/consumer/pull/PullSizeTest.java]: This test case has removed and add new unit test in store.

File: store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java [hunk 4]
```diff
@@ -1110,7 +1110,7 @@ private boolean isTheBatchFull(int sizePy, int maxMsgNums, int bufferTotal, int
             return false;
         }
 
-        if ((messageTotal + 1) >= maxMsgNums) {
```
Comment (Jaskey) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: I will suggest  replace all the condition of `messageTotal +1 > xxx` into 
    
    messageTotal >= xxx

Comment (lindzh) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: Yet,That a good idea to make code clean

Comment (vongosling) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: why change to -1 in your new change:-)

Comment (Jaskey) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: I will still suggest using `>=`

Comment (lindzh) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java]: Yes,this problem has been fixed to maxMsgNums <= messageTotal

**Comment by vongosling** [comment]
Now, LGTM

**Comment by lollipopjin** [comment]
+1

File: store/src/test/java/org/apache/rocketmq/store/DefaultMessageStoreTest.java [hunk 5]
```diff
@@ -45,19 +47,22 @@ public void init() throws Exception {
...
     }
 
+    public MessageStore buildMessageStore() throws Exception {
+        MessageStoreConfig messageStoreConfig = new MessageStoreConfig();
+        messageStoreConfig.setMapedFileSizeCommitLog(1024 * 1024 * 10);
+        messageStoreConfig.setMapedFileSizeConsumeQueue(1024 * 1024 * 10);
+        messageStoreConfig.setMaxHashSlotNum(10000);
+        messageStoreConfig.setMaxIndexNum(100 * 100);
+        messageStoreConfig.setFlushDiskType(FlushDiskType.ASYNC_FLUSH);
+        return new DefaultMessageStore(messageStoreConfig, new BrokerStatsManager("simpleTest"), new MyMessageArrivingListener(), new BrokerConfig());
```
Comment (lindzh) [inline store/src/test/java/org/apache/rocketmq/store/DefaultMessageStoreTest.java]: Indeed this is a must after testing,and this bug has been fix in PR https://github.com/apache/incubator-rocketmq/pull/141

**Comment by vongosling** [comment]
Thanks @lindzh, i have merged it. Please close this PR :-)

**Pull request closed without merging**